proposals/0000-lambda-layout.md

```diff
@@ -0,0 +1,353 @@
+
+forAll mkFold $ \(l0,l1,l2) -> do
+  let {- various bindings -}
+  \of | end0 && col1 <= col0 -> prs p s `shouldFailWith`
+        errFancy (getIndent l1 + g 1) (ii GT col0 col1)
+      | end1 && col2 <= col0 -> prs p s `shouldFailWith`
+        errFancy (getIndent l2 + g 2) (ii GT col0 col2)
+      | otherwise -> prs p s `shouldParse` (sbla, sblb, sblc)
+```
+
+caramia-0.7.2.2/src/Graphics/Caramia/Texture.hs:
+```Haskell
+return $ if
+    | result == GL_CLAMP_TO_EDGE -> Clamp
+    | result == GL_REPEAT -> Repeat
+    | otherwise -> error "getWrapping: unexpected wrapping mode."
+
+-- with -XMultiWayLambda and -XBlockArguments
+
+return \of
+    | result == GL_CLAMP_TO_EDGE -> Clamp
+    | result == GL_REPEAT -> Repeat
+    | otherwise -> error "getWrapping: unexpected wrapping mode."
+```
+
+red-black-record-2.1.0.3/lib/Data/RBR/Internal.hs
+```Haskell
+_prefixNS = \case
+    Left  l -> S l
+    Right x -> case x of Here fv -> Z @_ @v @start fv
+_breakNS = \case
+    Z x -> Right (Here x)
+    S x -> Left x
+
+-- with -XMultiWayLambda
+_prefixNS = \of
+    (Left  l) -> S l
+    (Right x) -> case x of Here fv -> Z @_ @v @start fv
+_breakNS = \of
+    (Z x) -> Right (Here x)
+    (S x) -> Left x
+```
+
+recursors-0.1.0.0/Control/Final.hs
+```Haskell
+map (\case PlainTV n    -> n
+           KindedTV n _ -> n) binders
+           
+-- With -XMultiWayLambda
+
+map (\of (PlainTV n)    -> n
+         (KindedTV n _) -> n) binders
+```
+
+roc-id-0.1.0.0/library/ROC/ID/Gender.hs
+```Haskell
+printGender :: Language -> Gender -> Text
+printGender = \case
+  English -> printGenderEnglish
+  Chinese -> printGenderChinese
+
+printGenderEnglish :: Gender -> Text
+printGenderEnglish = \case
+  Male   -> "Male"
+  Female -> "Female"
+
+printGenderChinese :: Gender -> Text
+printGenderChinese = \case
+  Male   -> "男性"
+  Female -> "女性"
+
+-- With -XMultiWayLambda - this makes use of the capability to have multiple parameters
+
+printGender :: Language -> Gender -> Text
+printGender = \of
+  English Male   -> "Male"
+  English Female -> "Female"
+  Chinese Male   -> "男性"
+  Chinese Female -> "女性"
+```
+
+## Effect and Interactions
+
+Enabling the extension enables users to use the suggested syntax. This obviates
+the need for `-XMultiWayIf` and `-XLambdaCase`.
+
+As `of` is already a keyword, no currently allowed syntax is stolen by this extension,
+and the behavior of no currently legal program would be changed with the extension
+enabled.
+
+## Costs and Drawbacks
+
+It is one additional syntactic construct to maintain, however the maintenance
+cost should be fairly low due to the similarity to already existing constructs.
+
+While this also means one additional construct to learn for beginners, the
+syntax is consistent with similar constructs in the existing language, and as
+such users might be surprised that a construct with these capabilities
+doesn't yet exist.
+
+## Alternatives
+
+ - The name of the extension could be something else.
```

Comment:
You can take this one out :slightly_smiling_face: . Unless you have another name you would like to record here. But otherwise, no need to state the obvious.